### DISCUSSION TITLE
Please improve DocFX documentation

### DISCUSSION THREAD

**Discussion by nathan-alden-sr** [dotnet/docfx#8336] [General]
It seems strange that a project whose purpose is to build documentation itself has such [poor documentation](https://dotnet.github.io/docfx/tutorial/docfx_getting_started.html). The DocFX tool and concept seem very cool and flexible; however, there is apparently no official documentation describing the architecture of DocFX--what each concept means, how they relate, etc. (documentation about the domain model of the tool)--nor is there detailed documentation about the `docfx.json` configuration file (yes, there is a JSON schema but that is too fine-grained to be of much use).

I was told there is a v3 of DocFX underway. Please consider greatly enhancing DocFX documentation to better introduce new folks to the tool and its model. Adoption will likely increase greatly if you do this.

**Comment by grokys** [comment]
PLEASE! I've just come to docfx again after several years and disappointed to find the documentation as obtuse as ever. @nathan-alden-sr correctly points to the `docfx.json` documentation being bad, but the documentation for theming is even worse.

**Comment by mdschweda** [comment]
Agreed. The state of the documentation is quite ironic for a documentation tool.

**Comment by AZ-X** [comment]
Quite agreed. I just narrow down to a sub project called ‘MarkdownLite’.  The codes inside this project are strange to me, while pattern’s names are superficial in everywhere.

**Comment by yufeih** [comment]
Thank you Nathan, these are all great feedbacks and I share the same feeling. There are so many areas to improve for a project whose whole purpose is to create documentation. I took some time to rework some of the docs to be more customer facing and hopefully it'll make it easier for people to understand and follow along.

There are much more improvements needed and we need the community's help to make the documentation better, so I'm converting this issue into a discussion for the community to collaboratively discuss and improve the product docs.

**Reply by killeriukas** [reply]
The documentation website has been changed like an hour ago. All the useful information has been deleted. Why?

This is an old documentation amount:
![image](https://user-images.githubusercontent.com/5539699/212229242-d304d9ef-ddbf-4bd7-a4ce-5ce257487dad.png)

This is a new documentation amount:
![image](https://user-images.githubusercontent.com/5539699/212229300-f2b796e1-793f-4bee-9219-0fb72f72620a.png)

Could you please at least return the old documentation? or provide an access to it? It's such a difficult tool to start using and now all the documentation has been removed too :(

EDIT: Just noticed that it is uploaded directly to this repo and can be read by selecting the tag. Wow... Got scared a little. Still - removing documentation is a big no no. I'd suggest you put it under obsolete tab until the new documentation is up to strach or better than the old.

**Reply by yufeih** [reply]
The biggest areas removed are: __Incremental Build__ and __Extensibility__. The reset of the essential docs are reorganized and kept around.

The reason for removing __Incremental Build__ is that we'd like to move away from the current approach of incremental build and instead focusing on improving the raw build performance. __Extensibility__ is another area we are looking at to approach differently, but yes as you've mentioned, keeping it under an obsolete tab is probably a good idea.

**Reply by KalleOlaviNiemitalo** [reply]
Some of <https://github.com/dotnet/docfx/blob/392eefac47ba4dd28e66493a6a5e804c6fc3c475/docs/spec/docfx_flavored_markdown.md> seems to be missing:

- Cross Reference (`xref:` URIs and `@`) linking to uid, and similar info in <https://github.com/dotnet/docfx/blob/392eefac47ba4dd28e66493a6a5e804c6fc3c475/docs/tutorial/links_and_cross_references.md>
- Section Definition -- but I suppose `<div>` can be used instead

Also <https://github.com/dotnet/docfx/blob/392eefac47ba4dd28e66493a6a5e804c6fc3c475/docs/tutorial/intro_overwrite_files.md> -- are overwrite files no longer supported?

**Reply by Stoffelche** [reply]
Somehow I got around with the documentation. Knew where to find somethings. And now it is all gone. All my bookmarks not working any more. <br>
The manual info is missing as well. Where can we find the info previously found at https://dotnet.github.io/docfx/tutorial/docfx.exe_user_manual.html?<br>
This kind of handling is not ok, should at least be announced somehow. The new doc should not replace the old docs, it should be supplemental.

**Reply by KalleOlaviNiemitalo** [reply]
The documentation of the `docfx.json` configuration file in <https://github.com/dotnet/docfx/blob/v2.60.0/docs/docs/config.md> omits at least the `xrefService`, `dest`, and `template` properties of `build`. <https://github.com/SchemaStore/schemastore/blob/8778c9f9ebeb448d0336209f9092fdc9ed6d97f2/src/schemas/json/docfx.json> does not describe all of those, either.

**Reply by Stoffelche** [reply]
At least one can use google cache to find part of the old documentation

**Reply by paulushub** [reply]
> At least one can use google cache to find part of the old documentation

Or simply generate it yourself.

**Reply by Stoffelche** [reply]
All the links found anywhere are broken with the old documentation is gone, like for example on this page:
https://microsoft.github.io/code-with-engineering-playbook/documentation/recipes/using-docfx-and-tools/

**Reply by killeriukas** [reply]
There was more documentation under Specifications tab as well. It was very helpful learning the nitty gritty stuff. Would be lovely if all of it was returned under the Obsolete tag as well. Thanks.

**Comment by tibitoth** [comment]
I'm really considering to move away from docfx to MKDocs Material. Maybe try to integrate docs generation from assembly into MKDocs but handwritten docs we will use MKDocs only.

**Reply by Stoffelche** [reply]
Are you generating the api documentation for c# projects with MKDocs as well? If so, how?

**Reply by tibitoth** [reply]
I've just done some investigation on this topic and sadly there is no easy way to achieve this because of this issue https://github.com/dotnet/docfx/issues/684 It seems like docfx do not create inmediate MD files and the process looks like this:  .dll => .xml => .yml? => .html
